libtenzir/builtins/formats/zeek_tsv.cpp

```diff
@@ -801,19 +801,29 @@ class zeek_tsv_printer final : public plugin_printer {
                                 args_.empty_field.value_or("(empty)"),
                                 args_.unset_field.value_or("-"),
                                 args_.disable_timestamp_tags};
-    return printer_instance::make([printer = std::move(printer)](
+    auto last_schema = std::make_shared<type>();
+    return printer_instance::make([last_schema = last_schema,
+                                   printer = std::move(printer)](
                                     table_slice slice) -> generator<chunk_ptr> {
+      if (slice.rows() == 0) {
+        co_yield {};
+        co_return;
+      }
       auto buffer = std::vector<char>{};
       auto out_iter = std::back_inserter(buffer);
       auto resolved_slice = flatten(resolve_enumerations(slice)).slice;
       auto input_schema = resolved_slice.schema();
       auto input_type = caf::get<record_type>(input_schema);
       auto array
         = to_record_batch(resolved_slice)->ToStructArray().ValueOrDie();
-      auto first = true;
+      auto should_print_close = static_cast<bool>(*last_schema);
+      auto first = std::exchange(*last_schema, input_schema) != input_schema;
```

Comment:
Incredibly hard to read...can we please simplify this?

Reply:
Would it be easier to read if I renamed `first` to `did_schema_change`?

Reply:
Oh, I meant the `std::exchange` part. This reads very cryptic. Can't we make this easier to read?

But yeah, renaming `should_print_close` could be `schema_changed`.

Reply:
I personally find the exchange to be quite readable, but I suppose it's something that one would need to get used to. I can move the swap to the line below, as in this particular case it doesn't make any difference.

Reply:
Unless there's a performance reason to do this, I would love to keep things simple for those with less deep C++ knowledge. That's my only motivation. And to be honest, I would have to look up the `std::exchange` idiom, even though that we would be know the 3rd time or so.

Reply:
I rewrote this to be a bit more readable.